server/src/main/java/org/opensearch/cluster/routing/allocation/allocator/BalancedShardsAllocator.java

```diff
@@ -531,6 +549,10 @@ void updateAllocationConstraint(String constraint, boolean enable) {
         void updateRebalanceConstraint(String constraint, boolean add) {
             this.rebalanceConstraints.updateRebalanceConstraint(constraint, add);
         }
+
+        void updatePrimaryConstraintThreshold(long primaryConstraintThreshold) {
+            this.primaryConstraintThreshold = primaryConstraintThreshold;
+        }
```

Comment:
What happens to the in-memory precomputed weights? Are they always created during the shard allocation loop?

Reply:
They are always computed again when the new shard allocation starts here in [decideAllocatedUnassigned](https://github.com/opensearch-project/OpenSearch/blob/c19cf88d266d15d6e8c69296cd0626099e562e38/server/src/main/java/org/opensearch/cluster/routing/allocation/allocator/LocalShardsBalancer.java#L945) function

Reply:
When the setting is updated, the new allocation uses the updated weights.